rs/system_api/src/lib.rs

```diff
@@ -3188,9 +3188,11 @@ impl SystemApi for SystemApiImpl {
                     // Access to this syscall not permitted.
                     Err(self.error_for("ic0_mint_cycles"))
                 } else {
-                    self.sandbox_safe_system_state
+                    let actually_minted = self
+                        .sandbox_safe_system_state
                         .mint_cycles(Cycles::from(amount))?;
-                    Ok(amount)
+                    // the actually minted amount cannot be larger than the argument which is a u64.
+                    Ok(actually_minted.low64())
```

Comment:
Hmm. Could it be system-dependent what we get here? Perhaps that's not quite right yet.

Reply:
I'd suggest that `sandbox_safe_system_state` returns Cycles that you try to decompose and trap if the amount exceeds 64-bit range.

Reply:
`actually_minted.into_parts().1` is exactly `actually_minted.low64()`, so I already decompose it correctly. 
I _could_ add a trap if `actually_minted.high64()` is not 0.

Reply:
> actually_minted.into_parts().1 is exactly actually_minted.low64(), so I already decompose it correctly.

I still believe it'd be more robust to use the Cycles type

Reply:
But `sandbox_safe_system_state.mint_cycles()` does return cycles - that's why I can call low64() on it.

Reply:
I see: I was confused by the question "Could it be system-dependent" which made be think that the return value is raw integers.